docs/ca/dev/monitoring/opentelemetry-instrumentation.md

```diff
@@ -0,0 +1,586 @@
+use Spryker\Service\Opentelemetry\Plugin\OpentelemetryMonitoringExtensionPlugin;
+
+class MonitoringDependencyProvider extends SprykerMonitoringDependencyProvider
+{
+    /**
+     * @return array<\Spryker\Service\MonitoringExtension\Dependency\Plugin\MonitoringExtensionPluginInterface>
+     */
+    protected function getMonitoringExtensions(): array
+    {
+        return [
+            new OpentelemetryMonitoringExtensionPlugin(),
+        ];
+    }
+}
+
+```
+
+
+You can call methods from Monitoring service, and they will be translated to OTel actions. Some methods act as placeholders because they're are not implemented in OTel, like `\Spryker\Service\Opentelemetry\Plugin\OpentelemetryMonitoringExtensionPlugin::markStartTransaction()` as transaction will start anyway.
+
+
+
+### Wire a console command
+
+Spryker is a large application, so manually creating all hooks is impractical. The `OpentelemetryGeneratorConsole` command automates hook generation for classes you want to cover with spans.
+
+```php
+<?php
+
+namespace Pyz\Zed\Console;
+
+...
+use Spryker\Zed\Opentelemetry\Communication\Plugin\Console\OpentelemetryGeneratorConsole;
+...
+
+class ConsoleDependencyProvider extends SprykerConsoleDependencyProvider
+{
+    protected function getConsoleCommands(Container $container): array
+    {
+        $commands = [
+            ...
+            new OpentelemetryGeneratorConsole(),
+            ...
+        ];
+
+        return $commands;
+    }
+}
+
+```
+
+Wire this console command into your install script to run on every deployment during the base container image build. Place it last in the `build` section, ensuring it runs after all code modifications and generation.
+
+```yaml
+sections:
+    build:
+        generate-open-telemetry:
+            command: 'vendor/bin/console open-telemetry:generate'
+```
+
+### Hooks generation configuration
+
+You can control instrumentation by configuring specific methods.
+
+`\Spryker\Zed\Opentelemetry\OpentelemetryConfig::getExcludedDirs()` defines directories to exclude from instrumentation. For example, you don't need spans from infrastructure code in traces. Several directories are excluded by default, review them in the module's vendor directory if you need to include any.  
+
+```php
+class OpentelemetryConfig extends AbstractBundleConfig
+{
+    // Traces for Monitoring module, OTel module and tests in existing module are not relevant for the monitoring, so those will be excluded. In the actual implementation you can see more directories.
+    public function getExcludedDirs(): array
+    {
+        return [
+        ...
+            'Monitoring',
+            'OpenTelemetry',
+            'tests',
+        ...
+        ];
+    }
+}
+```
+
+Use `\Spryker\Zed\Opentelemetry\OpentelemetryConfig::getExcludedSpans()` defines spans by name to exclude from instrumentation. This may be useful when spans you want to include and those you want to keep are in the same directory.
```

Comment:
```suggestion
`\Spryker\Zed\Opentelemetry\OpentelemetryConfig::getExcludedSpans()` defines spans by name to exclude from instrumentation. This may be useful when spans you want to include and those you want to keep are in the same directory.
```